### PR TITLE
Upgrade to CodeNarc 2.0

### PR DESCRIPTION
**Pull request by bmuskalla**
See [Changelog](https://github.com/CodeNarc/CodeNarc/blob/master/CHANGELOG.md)

### CODE REVIEW

**Comment by bmuskalla** [comment]
@bot-gradle test this

**Review by ljacomet — APPROVED** [review]

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc [hunk 1]
```diff
@@ -45,6 +45,7 @@ The API supporting the Java Toolchain feature in `org.gradle.jvm.toolchain` is n
 ==== Updates to default tool integration versions
 
 - JaCoCo has been updated to http://www.jacoco.org/jacoco/trunk/doc/changes.html[0.8.6].
+- CodeNarc has been updated to https://github.com/CodeNarc/CodeNarc/blob/v1.6.1/CHANGELOG.md[CodeNarc 2.0.0].
```
Comment (big-guy) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_6.adoc:48]: Don't you mean? https://github.com/CodeNarc/CodeNarc/blob/v2.0.0/CHANGELOG.md#version-200----oct-2020

File: subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CodeNarcPlugin.java [hunk 2]
```diff
@@ -37,7 +37,7 @@
  */
 public class CodeNarcPlugin extends AbstractCodeQualityPlugin<CodeNarc> {
 
-    public static final String DEFAULT_CODENARC_VERSION = "1.6.1";
+    public static final String DEFAULT_CODENARC_VERSION = "2.0.0";
```
Comment (big-guy) [inline subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CodeNarcPlugin.java:40]: When we make big jumps like this, we should also revisit CodenarcCoverage to see if we need to keep around the old version.  I would probably drop 1.2.1 and replace it with 1.6.1.

**Review by big-guy — CHANGES_REQUESTED** [review]
Thanks for doing this!

File: subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CodeNarcPlugin.java (hunk 2 above)
Comment (bmuskalla) [reply subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CodeNarcPlugin.java:40]: Thanks for catching

**Comment by bmuskalla** [comment]
@Mergifyio rebase

**Comment by bmuskalla** [comment]
@bot-gradle test this

**Pull request merged**